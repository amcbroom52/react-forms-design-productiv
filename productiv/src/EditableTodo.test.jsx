import { describe, it, expect, vi } from "vitest";

import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EditableTodo from "./EditableTodo";

const TODO = {
  id: '1',
  title: "title",
  description: "description",
  priority: 1
};

describe("Editable Todo", function () {
  it("renders without crashing", function () {
    render(<EditableTodo todo={TODO} />);
  });

  it("matches snapshot before pressing edit", function () {
    const { asFragment } = render(<EditableTodo todo={TODO} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot after pressing edit", function () {
    const { asFragment, container } = render(<EditableTodo todo={TODO} />);

    const editButton = container.querySelector(".EditableTodo-toggle");
    fireEvent.click(editButton);

    expect(asFragment()).toMatchSnapshot();
  });

  it("changes components properly", function () {
    const mockUpdate = vi.fn();

    const { container } = render(<EditableTodo todo={TODO} update={mockUpdate} />);

    //edit button before changing components
    let editButton = container.querySelector(".EditableTodo-toggle");
    expect(container).toContain(editButton);

    fireEvent.click(editButton);

    expect(container).not.toContain(editButton);

    //form component submit button
    const submitButton = container.querySelector(".NewTodoForm-addBtn");
    expect(container).toContain(submitButton);

    fireEvent.click(submitButton);

    //edit button after changing components
    editButton = container.querySelector(".EditableTodo-toggle");
    expect(container).toContain(editButton);

    expect(container).not.toContain(submitButton);
  });
});