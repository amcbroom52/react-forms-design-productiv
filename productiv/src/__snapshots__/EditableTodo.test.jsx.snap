// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Editable Todo > changes after editing 1`] = `
<DocumentFragment>
  <div
    class="EditableTodo"
  >
    <form
      class="NewTodoForm"
    >
      <div
        class="mb-3"
      >
        <input
          aria-label="Title"
          class="form-control"
          id="newTodo-title"
          name="title"
          placeholder="Title"
          value="title"
        />
      </div>
      <div
        class="mb-3"
      >
        <textarea
          aria-label="Description"
          class="form-control"
          id="newTodo-description"
          name="description"
          placeholder="Description"
        >
          description
        </textarea>
      </div>
      <div
        class="mb-3 d-flex justify-content-between"
      >
        <div
          class="w-75 d-flex justify-content-between"
        >
          <label
            class="d-inline-flex"
            for="newTodo-priority"
          >
            Priority:  
          </label>
          <select
            class="form-control form-control-sm d-inline-flex"
            id="newTodo-priority"
            name="priority"
          >
            <option
              value="1"
            >
              Ultra-Über
            </option>
            <option
              value="2"
            >
              Über
            </option>
            <option
              value="3"
            >
              Meh
            </option>
          </select>
        </div>
        <button
          class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
        >
          Gø!
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Editable Todo > matches snapshot after edit 1`] = `
<DocumentFragment>
  <div
    class="EditableTodo"
  >
    <form
      class="NewTodoForm"
    >
      <div
        class="mb-3"
      >
        <input
          aria-label="Title"
          class="form-control"
          id="newTodo-title"
          name="title"
          placeholder="Title"
          value="title"
        />
      </div>
      <div
        class="mb-3"
      >
        <textarea
          aria-label="Description"
          class="form-control"
          id="newTodo-description"
          name="description"
          placeholder="Description"
        >
          description
        </textarea>
      </div>
      <div
        class="mb-3 d-flex justify-content-between"
      >
        <div
          class="w-75 d-flex justify-content-between"
        >
          <label
            class="d-inline-flex"
            for="newTodo-priority"
          >
            Priority:  
          </label>
          <select
            class="form-control form-control-sm d-inline-flex"
            id="newTodo-priority"
            name="priority"
          >
            <option
              value="1"
            >
              Ultra-Über
            </option>
            <option
              value="2"
            >
              Über
            </option>
            <option
              value="3"
            >
              Meh
            </option>
          </select>
        </div>
        <button
          class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
        >
          Gø!
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Editable Todo > matches snapshot after editing 1`] = `
<DocumentFragment>
  <div
    class="EditableTodo"
  >
    <form
      class="NewTodoForm"
    >
      <div
        class="mb-3"
      >
        <input
          aria-label="Title"
          class="form-control"
          id="newTodo-title"
          name="title"
          placeholder="Title"
          value="title"
        />
      </div>
      <div
        class="mb-3"
      >
        <textarea
          aria-label="Description"
          class="form-control"
          id="newTodo-description"
          name="description"
          placeholder="Description"
        >
          description
        </textarea>
      </div>
      <div
        class="mb-3 d-flex justify-content-between"
      >
        <div
          class="w-75 d-flex justify-content-between"
        >
          <label
            class="d-inline-flex"
            for="newTodo-priority"
          >
            Priority:  
          </label>
          <select
            class="form-control form-control-sm d-inline-flex"
            id="newTodo-priority"
            name="priority"
          >
            <option
              value="1"
            >
              Ultra-Über
            </option>
            <option
              value="2"
            >
              Über
            </option>
            <option
              value="3"
            >
              Meh
            </option>
          </select>
        </div>
        <button
          class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
        >
          Gø!
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Editable Todo > matches snapshot after pressing edit 1`] = `
<DocumentFragment>
  <div
    class="EditableTodo"
  >
    <form
      class="NewTodoForm"
    >
      <div
        class="mb-3"
      >
        <input
          aria-label="Title"
          class="form-control"
          id="newTodo-title"
          name="title"
          placeholder="Title"
          value="title"
        />
      </div>
      <div
        class="mb-3"
      >
        <textarea
          aria-label="Description"
          class="form-control"
          id="newTodo-description"
          name="description"
          placeholder="Description"
        >
          description
        </textarea>
      </div>
      <div
        class="mb-3 d-flex justify-content-between"
      >
        <div
          class="w-75 d-flex justify-content-between"
        >
          <label
            class="d-inline-flex"
            for="newTodo-priority"
          >
            Priority:  
          </label>
          <select
            class="form-control form-control-sm d-inline-flex"
            id="newTodo-priority"
            name="priority"
          >
            <option
              value="1"
            >
              Ultra-Über
            </option>
            <option
              value="2"
            >
              Über
            </option>
            <option
              value="3"
            >
              Meh
            </option>
          </select>
        </div>
        <button
          class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
        >
          Gø!
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Editable Todo > matches snapshot before edit 1`] = `
<DocumentFragment>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3"
    >
      <div
        class="float-end text-sm-end"
      >
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
      <div
        class="Todo"
        id="1"
      >
        <div>
          <b>
            title
          </b>
           
          <small>
            Priority: 1
          </small>
        </div>
        <div>
          <small>
            description
          </small>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Editable Todo > matches snapshot before pressing edit 1`] = `
<DocumentFragment>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3"
    >
      <div
        class="float-end text-sm-end"
      >
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
      <div
        class="Todo"
        id="1"
      >
        <div>
          <b>
            title
          </b>
           
          <small>
            Priority: 1
          </small>
        </div>
        <div>
          <small>
            description
          </small>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
